paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc

```diff
@@ -131,44 +173,115 @@ void MainTest(const ProgramDesc& prog, int removed_nodes_num) {
   EXPECT_EQ(original_nodes_num - removed_nodes_num, current_nodes_num);
 }
 
+// check if scale_out is equal
+void EqualScaleTest(const ProgramDesc& prog, float scale) {
+  std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
+
+  // Init scope, as it is used in pass
+  auto place = paddle::platform::CPUPlace();
+  NaiveExecutor exe{place};
+  Scope scope;
+  exe.CreateVariables(prog, 0, true, &scope);
+
+  for (auto& v : variable_names) {
+    InitTensorHolder(&scope, place, v.c_str());
+  }
+
+  graph->SetNotOwned(kParamScopeAttr, &scope);
+
+  auto pass = PassRegistry::Instance().Get("cpu_quantize_squash_pass");
+
+  graph.reset(pass->Apply(graph.release()));
+
+  for (auto* node : graph->Nodes()) {
+    if (node->IsOp() && node->Op()->Type() == "conv2d") {
+      float scale_out = boost::get<float>(node->Op()->GetAttr("Scale_out"));
+      EXPECT_EQ(scale_out, scale);
+    }
+  }
+}
+
+// From Conv1->d->Dequant->e->Quant->f->Conv2
+// To Conv1->d->Conv2
 TEST(CpuQuantizeSquashPass, equal_scales) {
   auto scale = 1.2345f;
   auto use_mkldnn = true;
   // Remove 4 nodes: Dequant, Quant, e, f
   auto remove_nodes = 4;
-  MainTest(BuildProgramDesc(use_mkldnn, scale, scale), remove_nodes);
+  CountNodeTest(BuildProgramDesc(use_mkldnn, scale, scale), remove_nodes);
 
   use_mkldnn = !use_mkldnn;
```

Comment:
Testing with `use_mkldnn=false` should be dropped as quantization can be enabled only when `use_mkldnn=true`.
Here and below.